Bridges/HttpKernel.php

```diff
@@ -63,23 +64,81 @@ public function bootstrap($appBootstrap, $appenv)
      */
     public function onRequest(ReactRequest $request, ReactResponse $response)
     {
-        if (null !== $this->application) {
-            try {
-                $syRequest = new SymfonyRequest();
-                $syRequest->headers->replace($request->getHeaders());
-                $syRequest->setMethod($request->getMethod());
-                $syRequest->server->set('REQUEST_URI', $request->getPath());
-                $syRequest->server->set('SERVER_NAME', explode(':', $request->getHeaders()['Host'])[0]);
-
-                $syResponse = $this->application->handle($syRequest);
-                $this->application->terminate($syRequest, $syResponse);
-
-                $headers = array_map('current', $syResponse->headers->all());
-                $response->writeHead($syResponse->getStatusCode(), $headers);
-                $response->end($syResponse->getContent());
-            } catch (\Exception $e) {
-                //
+        if (null === ($app = $this->application)) {
+            return;
+        }
+
+        $content = '';
+        $headers = $request->getHeaders();
+        $contentLength = isset($headers['Content-Length']) ? (int) $headers['Content-Length'] : 0;
+
+        $request->on('data', function($data)
+            use ($request, $response, &$content, $contentLength)
+        {
+            // read data (may be empty for GET request)
+            $content .= $data;
+
+            // handle request after receive
+            if (strlen($content) >= $contentLength) {
```

Comment:
Regarding the http specification, `content-length` is not necessary. what if the sender doesn't send it ($contentLength = 0)? guess we should read then till the end, right?